common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -3865,6 +3865,8 @@ public static enum ConfVars {
         "get old behavior, if desired. See, test-case in patch for HIVE-6689."),
     HIVE_LINEAGE_INFO("hive.lineage.hook.info.enabled", false,
         "Whether Hive provides lineage information to hooks."),
+    HIVE_LINEAGE_STATEMENT_FILTER("hive.lineage.statement.filter", "All",
```

Comment:
I don't know if we have a rule about it or not. But can the default value be lowercase? 
I would love to avoid future confusion about if is that All, all or ALL. 

Also, could you please put some description and and example values there that can help us to understand what that config value exactly does?

Reply:
The comparison is not case sensitive to make this more felxible
```
if (ALL.equalsIgnoreCase(valueText)) {
```
```
if (NONE.equalsIgnoreCase(valueText)) {
```
```
HiveOperation enumValue = EnumUtils.getEnumIgnoreCase(HiveOperation.class, valueText);
```

I added an upper case default value because other values coming from HiveOperation are in upper case. There are other configs with upper case values so I think it is not an exception.
I also added the missing description.